packages/flutter_tools/lib/src/linux/build_linux.dart

```diff
@@ -38,10 +42,38 @@ Future<void> buildLinux(LinuxProject linuxProject, BuildInfo buildInfo, {String
   final StringBuffer buffer = StringBuffer('''
 # Generated code do not commit.
 export FLUTTER_ROOT=${Cache.flutterRoot}
-export TRACK_WIDGET_CREATION=${buildInfo?.trackWidgetCreation == true}
 export FLUTTER_TARGET=$target
 export PROJECT_DIR=${linuxProject.project.directory.path}
 ''');
+  if (buildInfo.dartDefines?.isNotEmpty ?? false) {
+    buffer.write('export DART_DEFINES=');
+    buffer.writeln(buildInfo.dartDefines.join(','));
+  }
+  if (buildInfo.dartObfuscation != null) {
+    buffer.writeln('export DART_OBFUSCATION=${buildInfo.dartObfuscation}');
+  }
+  if (buildInfo.extraFrontEndOptions?.isNotEmpty ?? false) {
+    buffer.write('export EXTRA_FRONT_END_OPTIONS=');
+    buffer.writeln(buildInfo.extraFrontEndOptions.join(','));
+  }
+  if (buildInfo.extraGenSnapshotOptions?.isNotEmpty ?? false) {
+    buffer.write('export EXTRA_GEN_SNAPSHOT_OPTIONS=');
+    buffer.writeln(buildInfo.extraGenSnapshotOptions.join(','));
+  }
+  if (buildInfo.splitDebugInfoPath != null) {
+    buffer.writeln('export SPLIT_DEBUG_INFO=${buildInfo.splitDebugInfoPath}');
+  }
+  if (buildInfo.trackWidgetCreation != null) {
+    buffer.writeln(
+      'export TRACK_WIDGET_CREATION=${buildInfo.trackWidgetCreation}',
+    );
+  }
+  if (buildInfo.treeShakeIcons != null) {
+    buffer.writeln(
+      'export TREE_SHAKE_ICONS=${buildInfo.treeShakeIcons}',
+    );
+  }
```

Comment:
Since the format, but not conceptual content, of this file will change in the switch to cmake, it would be nice to build this as a dictionary and then do an iterator over it at the end to write the `export` lines. That way switching to cmake won't have to touch all these lines again.

(And by a happy coincidence, that happens to be the same change needed to make it easy to share the code with Windows 😉)

Reply:
Yeah, good call :)

Done!